src/portable/portable.c

```diff
@@ -249,8 +262,21 @@ static int extract_now(
                         }
 
                         if (socket_fd >= 0) {
+#if HAVE_SELINUX
+                                /* The units will be copied on the host's filesystem, so if they had a SELinux label
+                                 * we have to preserve it. Copy it out so that it can be applied later. */
+                                _cleanup_freecon_ char *con = NULL;
+
+                                r = fgetfilecon(fd, &con);
+                                if (r < 0 && errno != ENODATA)
+                                        return log_debug_errno(errno, "Failed to get SELinux file context from '%s': %m", de->d_name);
```

Comment:
i think we should treat these selinux errors graacefully: if we can't propagate the label, then log, but proceed and let the default label on the inode